### PR TITLE
Using filesystem path 2 [ANT-1999]

### PR DESCRIPTION
**Pull request by payetvin**
following #2435

### CODE REVIEW

File: src/libs/antares/paths/include/antares/paths/list.h [hunk 1]
```diff
@@ -178,17 +178,6 @@ class PathList
...
-    size_t sizeOnDisk(const StringT& sourceFolder) const
-    {
-        if (item.empty())
-        {
-            return 0;
-        }
-        pTmp = sourceFolder;
-        return (!pTmp) ? 0 : internalSizeOnDisk();
-    }
-
```
Comment (flomnes) [inline src/libs/antares/paths/include/antares/paths/list.h:191]: Note to other readers: this function is unused

File: src/libs/antares/paths/list.cpp [hunk 2]
```diff
@@ -41,23 +41,6 @@ void PathList::clear()
...
-    {
-        if (!(i->second.options & pathListOptFolder))
-        {
-            buffer.clear() << pTmp << SEP << i->first;
-            size += (size_t)IO::File::Size(pTmp);
-        }
-    }
-    return size;
-}
-
```
Comment (flomnes) [inline src/libs/antares/paths/list.cpp:60]: Note: this function is unused.

File: src/libs/antares/paths/list.cpp [hunk 3]
```diff
@@ -70,6 +53,7 @@ size_t PathList::totalSizeInBytes() const
     return size;
 }
 
+// TODO VP: remove with tools
```
Comment (flomnes) [inline src/libs/antares/paths/list.cpp:56]:
```suggestion
 [[deprecated( "This function will be removed, don't use it (used by : tools)")]] 
```

File: src/libs/antares/study/include/antares/study/output.h [hunk 4]
```diff
@@ -34,6 +34,8 @@ namespace Antares
 {
 namespace Data
 {
+
+// TODO VP: remove with GUI
```
Comment (flomnes) [inline src/libs/antares/study/include/antares/study/output.h:38]:
```suggestion

[[deprecated ("Remove with GUI")]]
```

File: src/libs/antares/study/include/antares/study/parts/solar/prepro.h [hunk 5]
```diff
@@ -52,7 +52,7 @@ class Prepro
     /*!
     ** \brief Solar data from a folder
     */
-    bool loadFromFolder(const AnyString& folder);
+    bool loadFromFolder(const std::string& folder);
```
Comment (flomnes) [inline src/libs/antares/study/include/antares/study/parts/solar/prepro.h]:
```suggestion
    bool loadFromFolder(const std::filesystem::path& folder);
```

File: src/libs/antares/study/include/antares/study/parts/wind/prepro.h [hunk 6]
```diff
@@ -49,7 +49,7 @@ class Prepro
     /*!
     ** \brief Load data from a folder
     */
-    bool loadFromFolder(const AnyString& folder);
+    bool loadFromFolder(const std::string& folder);
```
Comment (flomnes) [inline src/libs/antares/study/include/antares/study/parts/wind/prepro.h]:
```suggestion
    bool loadFromFolder(const std::filesystem::path& folder);
```

File: src/libs/antares/study/load.cpp [hunk 7]
```diff
@@ -238,40 +231,32 @@ bool Study::internalLoadCorrelationMatrices(const StudyLoadOptions& options)
     if (!options.loadOnlyNeeded || timeSeriesLoad & parameters.timeSeriesToRefresh
         || timeSeriesLoad & parameters.timeSeriesToGenerate)
     {
-        buffer.clear() << folderInput << SEP << "load" << SEP << "prepro" << SEP
-                       << "correlation.ini";
-        preproLoadCorrelation.loadFromFile(*this, buffer);
+        fs::path loadPath = folderInput / "load" / "prepro" / "correlation.ini";
+        preproLoadCorrelation.loadFromFile(*this, loadPath.string());
```
Comment (flomnes) [inline src/libs/antares/study/load.cpp:235]:
```suggestion
        preproLoadCorrelation.loadFromFile(*this, loadPath);
```

File: src/libs/antares/study/load.cpp [hunk 8]
```diff
@@ -238,40 +231,32 @@ bool Study::internalLoadCorrelationMatrices(const StudyLoadOptions& options)
...
 
     // Solar
     if (!options.loadOnlyNeeded || timeSeriesSolar & parameters.timeSeriesToRefresh
         || timeSeriesSolar & parameters.timeSeriesToGenerate)
     {
-        buffer.clear() << folderInput << SEP << "solar" << SEP << "prepro" << SEP
-                       << "correlation.ini";
-        preproSolarCorrelation.loadFromFile(*this, buffer);
+        fs::path solarPath = folderInput / "solar" / "prepro" / "correlation.ini";
+        preproSolarCorrelation.loadFromFile(*this, solarPath.string());
```
Comment (flomnes) [inline src/libs/antares/study/load.cpp:243]:
```suggestion
        preproSolarCorrelation.loadFromFile(*this, solarPath);
```

File: src/libs/antares/study/parts/hydro/container.cpp [hunk 9]
```diff
@@ -183,7 +182,7 @@ bool PartHydro::LoadFromFolder(Study& study, const AnyString& folder)
               }
               else
               {
-                  ret = area.hydro.LoadDailyMaxEnergy(folder, area.id) && ret;
+                  ret = area.hydro.LoadDailyMaxEnergy(folder.string(), area.id) && ret;
```
Comment (flomnes) [inline src/libs/antares/study/parts/hydro/container.cpp:184]:
```suggestion
                  ret = area.hydro.LoadDailyMaxEnergy(folder, area.id) && ret;
```

File: src/libs/antares/study/parts/hydro/container.cpp [hunk 10]
```diff
@@ -192,36 +191,38 @@ bool PartHydro::LoadFromFolder(Study& study, const AnyString& folder)
...
           }
 
-          buffer.clear() << folder << SEP << "common" << SEP << "capacity" << SEP
-                         << "creditmodulations_" << area.id << '.' << study.inputExtension;
-          ret = area.hydro.creditModulation.loadFromCSVFile(buffer,
+          fs::path capacityPath = folder / "common" / "capacity";
+
+          std::string creditId = "creditmodulations_" + area.id + ".txt";
+          fs::path creditPath = capacityPath / creditId;
+          ret = area.hydro.creditModulation.loadFromCSVFile(creditPath.string(),
```
Comment (flomnes) [inline src/libs/antares/study/parts/hydro/container.cpp:197]:
```suggestion
          ret = area.hydro.creditModulation.loadFromCSVFile(creditPath,
```

File: src/libs/antares/study/parts/wind/prepro.cpp [hunk 11]
```diff
@@ -43,7 +35,7 @@ Prepro::~Prepro()
 {
 }
 
-bool Prepro::loadFromFolder(const AnyString& folder)
+bool Prepro::loadFromFolder(const std::string& folder)
```
Comment (flomnes) [inline src/libs/antares/study/parts/wind/prepro.cpp]:
```suggestion
bool Prepro::loadFromFolder(const std::filesystem::path& folder)
```

File: src/libs/antares/study/xcast/xcast.cpp [hunk 12]
```diff
@@ -246,41 +248,41 @@ bool XCast::loadFromFolder(const AnyString& folder)
...
     // It would be better to load the matrices after the settings in order to
     // fix invalid data
 
     // Coefficients
-    buffer.clear() << folder << SEP << "data.txt";
+    fs::path p = folder / "data.txt";
 
     // Performing normal loading
-    ret = data.loadFromCSVFile(buffer, (uint)dataMax, 12, Matrix<>::optFixedSize, &readBuffer)
+    ret = data.loadFromCSVFile(p.string(), (uint)dataMax, 12, Matrix<>::optFixedSize, &readBuffer)
```
Comment (flomnes) [inline src/libs/antares/study/xcast/xcast.cpp:273]:
```suggestion
    ret = data.loadFromCSVFile(p, (uint)dataMax, 12, Matrix<>::optFixedSize, &readBuffer)
```

File: src/solver/simulation/solver_utils.cpp [hunk 13]
```diff
@@ -145,7 +142,7 @@ void annualCostsStatistics::writeUpdateTimes(IResultWriter& writer) const
     buffer << "MAX (ms) : " << updateTime.costMax << "\n";
 
     std::string s = buffer.str();
-    writer.addEntryFromBuffer(updateTimeFilename, s);
+    writer.addEntryFromBuffer(updateTimeFilename.string(), s);
```
Comment (flomnes) [inline src/solver/simulation/solver_utils.cpp]:
```suggestion
    writer.addEntryFromBuffer(updateTimeFilename, s);
```

File: src/solver/simulation/solver_utils.cpp [hunk 14]
```diff
@@ -164,7 +161,7 @@ void annualCostsStatistics::writeOptimizationTimeToOutput(IResultWriter& writer)
     buffer << "MAX (ms) : " << optimizationTime2.costMax << "\n";
 
     std::string s = buffer.str();
-    writer.addEntryFromBuffer(optimizationTimeFilename, s);
+    writer.addEntryFromBuffer(optimizationTimeFilename.string(), s);
```
Comment (flomnes) [inline src/solver/simulation/solver_utils.cpp]:
```suggestion
    writer.addEntryFromBuffer(optimizationTimeFilename, s);
```

File: src/solver/variable/include/antares/solver/variable/container.hxx [hunk 15]
```diff
@@ -254,7 +251,7 @@ void List<NextT>::buildSurveyReport(SurveyResults& results,
 
     // The new filename
     results.data.filename.clear();
-    results.data.filename << results.data.output << SEP;
+    results.data.filename << std::filesystem::path(std::string(results.data.output)) / "";
```
Comment (flomnes) [inline src/solver/variable/include/antares/solver/variable/container.hxx]: This is ugly...

File: src/solver/variable/include/antares/solver/variable/surveyresults/reportbuilder.hxx [hunk 16]
```diff
@@ -354,9 +356,11 @@ private:
 
                 logs.info() << "Exporting results : " << area.name << " :: " << cluster->name();
                 // The new output
-                results.data.output.clear();
-                results.data.output << results.data.originalOutput << SEP << "areas" << SEP
-                                    << area.id << SEP << "thermal" << SEP << cluster->id();
+                std::filesystem::path path = std::string(results.data.originalOutput);
+                path /= std::filesystem::path("areas") / area.id.to<std::string>() / "thermal"
+                        / cluster->id();
```
Comment (flomnes) [inline src/solver/variable/include/antares/solver/variable/surveyresults/reportbuilder.hxx:361]: If you have a `typedef`, you can just use
```suggestion
                path /= fs("areas") / area.id.to<std::string>() / "thermal"
                        / cluster->id();
```

**Review by flomnes — CHANGES_REQUESTED** [review]
Good job !
Please change the following
- `loadFromFolder(const std::string&, ...)` => `loadFromFolder(const std::filesystem::path&, ...)`
- same for `saveToFolder` ?
- same for `addFromBuffer`
- You may use attribute `[[deprecated]]`

File: src/solver/variable/include/antares/solver/variable/surveyresults/reportbuilder.hxx (hunk 16 above)
Comment (payetvin) [reply src/solver/variable/include/antares/solver/variable/surveyresults/reportbuilder.hxx:361]: It's a hxx file, i don't use typedef

File: src/solver/variable/include/antares/solver/variable/container.hxx (hunk 15 above)
Comment (payetvin) [reply src/solver/variable/include/antares/solver/variable/container.hxx]: I agree, but it avoids intermediate variables needed to convert from yuni string

**Review by flomnes — APPROVED** [review]

**Pull request merged**